### PR TITLE
[SVACE] Fix SVACE and covertiy issue

### PR DESCRIPTION
**Pull request by songgot**
- Fix covertiy and SVACE issue

**Self evaluation:**
1. Build test: [*]Passed [ ]Failed []Skipped
2. Run test: [*]Passed [ ]Failed []Skipped

### CODE REVIEW

File: gst/datarepo/gstdatareposink.c [hunk 1]
```diff
@@ -670,7 +670,7 @@ __write_json (JsonObject * object, const gchar * filename)
   json_generator_set_root (generator, root);
   json_generator_set_pretty (generator, TRUE);
   ret = json_generator_to_file (generator, filename, &error);
-  if (!ret) {
+  if (!ret && error) {
```
Comment (jaeyun-jung) [inline gst/datarepo/gstdatareposink.c:673]: plz check condition (!ret || error)

Comment (songgot) [reply gst/datarepo/gstdatareposink.c:673]: Thank you.
ret == false and error !=NULL

**Review by jaeyun-jung — APPROVED** [review]

File: gst/datarepo/gstdatareposink.c (hunk 1 above)
Comment (songgot) [reply gst/datarepo/gstdatareposink.c]: Thank you. error is removed

**Review by again4you — APPROVED** [review]
LGTM!

**Comment by songgot** [comment]
Thank you.

**Pull request merged**